solver/src/main/java/org/chocosolver/solver/expression/continuous/relational/PropEquation.java

```diff
@@ -0,0 +1,234 @@
+        else
+            return (this.exp.getLB() <= this.cste.getUB() && this.exp.getUB() >= this.cste.getLB());
+    }
+
+    protected boolean bc(RealVar var, CArExpression[] wx, CArExpression[] wox) throws ContradictionException {
+        int[] depths = new int[this.boxConsistencyDepth * 2];
+        int depth = 0;
+        int idx = 0;
+        boolean fin = false;
+
+        double leftB = 0, rightB = 0;
+        double[] oldValue = {var.getLB(), var.getUB()};
+
+        tighten(wox);
+
+        // Left bound !
+        while (!fin) {
+            if (not_inconsistent(wx)) {
+                if (this.boxConsistencyDepth <= depth) {
+                    leftB = var.getLB();
+                    rightB = var.getUB(); // Valeur provisoire
+                    fin = true;
+                } else {
+                    RealInterval left = RealUtils.firstHalf(var);
+                    RealInterval right = RealUtils.secondHalf(var);
+
+                    var.silentlyAssign(left);
+                    depth++;
+                    unexplored[idx] = right;
+                    depths[idx] = depth;
+                    idx++;
+                }
+            } else if (idx != 0) {
+                var.silentlyAssign(unexplored[--idx]);
+                depth = depths[idx];
+            } else {
+                return false;
+            }
+        }
+
+        // Reversing not explored intervals (in order to avoid to check already checked parts of the search space.
+
+        RealInterval[] tmp1 = new RealInterval[this.boxConsistencyDepth * 2];
+        int[] tmp2 = new int[this.boxConsistencyDepth * 2];
+
+        for (int i = 0; i < idx; i++) {
+            int j = idx - i - 1;
+            tmp1[i] = unexplored[j];
+            tmp2[i] = depths[j];
+        }
+
+        unexplored = tmp1;
+        depths = tmp2;
+
+        // Right bound if needed
+        if (idx != 0) {
+            var.silentlyAssign(unexplored[--idx]);
+            depth = depths[idx];
+            fin = false;
+
+            while (!fin) {
+                if (not_inconsistent(wx)) {
+                    if (this.boxConsistencyDepth <= depth) {
+                        rightB = var.getUB();
+                        fin = true;
+                    } else {
+                        RealInterval left = RealUtils.firstHalf(var);
+                        RealInterval right = RealUtils.secondHalf(var);
+
+                        var.silentlyAssign(right);
+                        depth++;
+                        unexplored[idx] = left;
+                        depths[idx] = depth;
+                        idx++;
+                    }
+                } else if (idx != 0) {
+                    var.silentlyAssign(unexplored[--idx]);
+                    depth = depths[idx];
+                } else {
+                    fin = true;
+                }
+            }
+        }
+
+        // Propagation
+        var.silentlyAssign(oldValue[0], oldValue[1]);
```

Comment:
Should be removed

Reply:
No it shouldn’t since this call ensures that the initial values are restored. Indeed, the two previous while-loop change these bounds abruptly to detect inconsistency.